consensus/src/dag/bootstrap.rs

```diff
@@ -472,21 +543,30 @@ impl DagBootstrapper {
         mut dag_rpc_rx: Receiver<Author, IncomingDAGRequest>,
         mut shutdown_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
+        info!(
+            LogSchema::new(LogEvent::Start),
+            epoch = self.epoch_state.epoch,
+        );
+
         let (base_state, handler, fetch_service) = self.full_bootstrap();
+
         let mut mode = Mode::Active(ActiveMode {
             handler,
             fetch_service,
             base_state,
+            buffer: Vec::new(),
         });
         loop {
             select! {
                 biased;
                 Ok(ack_tx) = &mut shutdown_rx => {
                     let _ = ack_tx.send(());
+                    info!(LogSchema::new(LogEvent::Shutdown), "shutdown complete");
                     return;
                 },
                 Some(next_mode) = mode.run(&mut dag_rpc_rx, &self) => {
-                    mode = next_mode
+                    info!(LogSchema::new(LogEvent::ModeTransition), next_mode = %next_mode);
+                    mode = next_mode;
```

Comment:
for completeness, probably worth adding a complete branch => break;

Reply:
hmm, I am using tokio::select and it panics if the branches exhaust. Do you prefer future::select! for some reason?

Reply:
it’s “else” in tokio select?

Reply:
I don't think so. It also runs if the patterns don't match right? We don't expect that shutdown match to anything other than Ok(), otherwise, it's better to panic than handle the case and return?

Reply:
yes it runs when patterns don’t match, it’s just defensive programming in case that shutdown channel was dropped for some reason it can still return, but panic might be okay too. i thought it’d spin loop